### ISSUE TITLE
JTA not supported

### ISSUE THREAD

**Issue by rmannibucau** [Impetus/kundera#905]
When using kundera in container mode (see PersistenceProvider) and JTA mode as required by EE spec, you need to enforce kundera JTA transaction implementation which is not an option in an EE server, kundera should be able to register with the running transaction synchronization registry to participate in the JTA transaction instead.

**Comment by cristof** [comment]
+1 from me.
I guess it's the same issue reported at #651 .

**Comment by rmannibucau** [comment]
Not sure it is fully the same (blocker vs spec issue) but outcome would likely fix both yes.

**Comment by cristof** [comment]
You're right about the differences. Would the fix suggested in https://github.com/impetus-opensource/Kundera/issues/651#issuecomment-63784324 attend the specification?
I think Kundera should make transparent to user the internal usage of the running transaction without a previous configuration.

**Comment by rmannibucau** [comment]
Not only. You still need to integrate with the transaction synchonization to be able to rollback with jta otherwise you passthrough but not support jta

**Comment by cristof** [comment]
Sure. As I said in #651 , it's a critical feature for CMT transactions. Last time I had a look on source of kundera, the rollback part kept me from proposing a patch. Thanks, @rmannibucau. Hope kundera guys can help out with the issue. It would be great for their project.
